src/main/java/group1/mips_simulator/components/instructionParts/Instruction.java

```diff
@@ -51,6 +52,24 @@ public String toString_Binary() {
         return result.toString();
     }
 
+
+    public short toShort(){
+        // generates an int from string of self,
+        // returns a short casting of this value
+        int rvalue = 0;
+        char[] instructions = toString_Binary().toCharArray();
+        for(int i = 0; i < instructions.length; i++){
+            if(instructions[instructions.length - i - 1] == '1')
+                rvalue |= 1<<i;
```

Comment:
Nice. Bit manipulation usually scares me. But this seems ok. Especially if you've got some unit tests here.

Reply:
🦾